packages/lib/src/core/Analytics/Analytics.ts

```diff
@@ -1,74 +1,82 @@
 import logEvent from '../Services/analytics/log-event';
-import postTelemetry from '../Services/analytics/post-telemetry';
 import collectId from '../Services/analytics/collect-id';
-import EventsQueue from './EventsQueue';
 import { CoreOptions } from '../types';
+import CAEventsQueue, { EQObject } from './CAEventsQueue';
+import { ANALYTICS_ACTION, AnalyticsInitialEvent, AnalyticsObject } from './types';
+import { ANALYTICS_ACTION_ERROR, ANALYTICS_ACTION_LOG, ANALYTICS_PATH } from './constants';
+import { debounce } from '../../components/internal/Address/utils';
+import { AnalyticsModule } from '../../components/types';
 
-export type AnalyticsProps = Pick<CoreOptions, 'loadingContext' | 'locale' | 'clientKey' | 'analytics' | 'amount'>;
+export type AnalyticsProps = Pick<CoreOptions, 'loadingContext' | 'locale' | 'clientKey' | 'analytics' | 'amount' | 'analyticsContext'>;
 
-class Analytics {
-    private static defaultProps = {
+let _checkoutAttemptId = null;
+
+const Analytics = ({ loadingContext, locale, clientKey, analytics, amount, analyticsContext }: AnalyticsProps) => {
```

Comment:
My thinking on this was why use a class when you don't need one?
Classes are something of an artificial construct for javascript and add add weight to the bundle.

Reply:
Sorry, this is actually called clojure 🤓

Not sure what do you mean by saying that classes are an artificial construct. Classes are supported natively since a long time! And in terms of weight - I don't think they are very much different after all. I didn't run the test by myself, but I couldn't find anything saying that this is a big deal.

I agree that is better for React.Components to be moved to functional components for sure - that is the "new" standard of writing UI components. But that does not mean that we are forced to write other parts of the code base (which are not Preact components) using functional approach.

My main point is: we generally don't use clojures in our project to write JS modules. We usually leverage Typescript/Classes or simple helper functions. Do we want to start introducing another way of writing/encapsulating objects/modules in the codebase? All Core modules are class-based - do we want to rewrite this one and make it different than the others, following another pattern? I'd rather avoid that and keep it as we are doing currently. If we want to introduce another way of writing modules, then we should discuss internally our coding guidelines and align on this.